modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/java/SourceModelLoaderDelegate.java

```diff
@@ -104,10 +123,63 @@ void declareMessageSources(ExtensionDeclarer extensionDeclarer,
       declareSourceCallbackParameters(parser.getOnTerminateCallbackParser(), sourceDeclarer::onTerminate);
       declareSourceCallbackParameters(parser.getOnBackPressureCallbackParser(), sourceDeclarer::onBackPressure);
 
+      declareClusterSupport(sourceDeclarer, parser.getSourceClusterSupport());
+      declareBackPressureSupport(extensionDeclarer, sourceDeclarer, parser.getBackPressureStrategyModelProperty());
+
       sourceDeclarers.put(parser, sourceDeclarer);
     }
   }
 
+  private void declareClusterSupport(SourceDeclarer sourceDeclarer, Optional<SourceClusterSupport> sourceClusterSupport) {
+    boolean runsOnPrimaryNodeOnly;
+    switch (sourceClusterSupport.orElse(SourceClusterSupport.DEFAULT_ALL_NODES)) {
+      case DEFAULT_PRIMARY_NODE_ONLY:
+        runsOnPrimaryNodeOnly = false;
+        addPrimaryNodeParameter(sourceDeclarer.getDeclaration(), true);
+        break;
+      case DEFAULT_ALL_NODES:
+        runsOnPrimaryNodeOnly = false;
+        addPrimaryNodeParameter(sourceDeclarer.getDeclaration(), false);
+        break;
+      case NOT_SUPPORTED:
+        runsOnPrimaryNodeOnly = true;
+        break;
+      default:
+        runsOnPrimaryNodeOnly = true;
+    }
+    sourceDeclarer.runsOnPrimaryNodeOnly(runsOnPrimaryNodeOnly);
+  }
+
+  private void declareBackPressureSupport(ExtensionDeclarer extensionDeclarer, SourceDeclarer sourceDeclarer,
```

Comment:
this should still happen on an enricher, only that one that's fixed in the ExtensionModelFactory and not java specific

Reply:
So in the parsing I add the model property, and consume it in the enricher?

What about the cluster support, does it also need to still be added in an enricher for all extensions?